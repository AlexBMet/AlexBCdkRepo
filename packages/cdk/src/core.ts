export const generateResourcePrefix = (uniquePrefix: string, environment: string, region: string) => {
	return `${uniquePrefix}-${environment}-${region}`
};
