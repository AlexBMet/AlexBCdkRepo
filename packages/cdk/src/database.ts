import {AttributeType, BillingMode, Table} from '@aws-cdk/aws-dynamodb';
import {CfnParameter, Construct, RemovalPolicy, Stack, Tag} from '@aws-cdk/core';

export class Database extends Stack {

	constructor(scope: Construct, id: string, props: {}) {
		super(scope, id, props);

		const STACK = Stack.of(this);
		const ENVIRONMENT = new CfnParameter(this, 'Environment', {
			allowedValues: ['dev', 'ci', 'stg', 'prod'],
			description: 'TBC',
			type: 'String'
		});
		const SERVICE_CODE = new CfnParameter(this, 'ServiceCode', {
			type: 'String',
			description: 'TBC'
		});
		const SERVICE_NAME = new CfnParameter(this, 'ServiceName', {
			type: 'String',
			description: 'TBC'
		});
		const SERVICE_OWNER = new CfnParameter(this, 'ServiceOwner', {
			type: 'String',
			description: 'TBC'
		});
		const UNIQUE_PREFIX = new CfnParameter(this, 'UniquePrefix', {
			type: 'String',
			description: 'TBC'
		});
		const PREFIX = `${UNIQUE_PREFIX.value}-${ENVIRONMENT.value}-${STACK.region}`;

		Tag.add(this, 'Environment', `${ENVIRONMENT.value}`);
		Tag.add(this, 'ServiceCode', `${SERVICE_CODE.value}`);
		Tag.add(this, 'ServiceName', `${SERVICE_NAME.value}`);
		Tag.add(this, 'ServiceOwner', `${SERVICE_OWNER.value}`);

		new Table(this, 'Database', {
			billingMode: BillingMode.PAY_PER_REQUEST,
			partitionKey: {
				name: 'id',
				type: AttributeType.STRING
			},
			removalPolicy: RemovalPolicy.DESTROY,
			tableName: `${PREFIX}-table`,
		});
	}
}
