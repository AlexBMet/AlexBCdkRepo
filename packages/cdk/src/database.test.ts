import { Database } from './database';
import { Stack } from '@aws-cdk/core';
import { SynthUtils } from '@aws-cdk/assert'; // tslint:disable-line:no-implicit-dependencies

describe('[database.ts] unit tests', () => {
	describe('[stack]', () => {
		it('must create a DynamoDB table', async () => {
			const stack = new Database(new Stack(), 'DatabaseStack', {});
			expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
		});
	});
});
