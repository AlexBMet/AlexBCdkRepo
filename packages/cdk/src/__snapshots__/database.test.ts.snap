// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[database.ts] unit tests [stack] must create a DynamoDB table 1`] = `
Object {
  "Parameters": Object {
    "Environment": Object {
      "AllowedValues": Array [
        "dev",
        "ci",
        "stg",
        "prod",
      ],
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceCode": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceName": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceOwner": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "UniquePrefix": Object {
      "Description": "TBC",
      "Type": "String",
    },
  },
  "Resources": Object {
    "DatabaseB269D8BB": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "TableName": Object {
          "Fn::Join": Array [
            "",
            Array [
              Object {
                "Ref": "UniquePrefix",
              },
              "-",
              Object {
                "Ref": "Environment",
              },
              "-",
              Object {
                "Ref": "AWS::Region",
              },
              "-table",
            ],
          ],
        },
        "Tags": Array [
          Object {
            "Key": "Environment",
            "Value": Object {
              "Ref": "Environment",
            },
          },
          Object {
            "Key": "ServiceCode",
            "Value": Object {
              "Ref": "ServiceCode",
            },
          },
          Object {
            "Key": "ServiceName",
            "Value": Object {
              "Ref": "ServiceName",
            },
          },
          Object {
            "Key": "ServiceOwner",
            "Value": Object {
              "Ref": "ServiceOwner",
            },
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
